codec/codec-jackson/src/main/java/com/alipay/sofa/rpc/codec/jackson/JacksonHelper.java

```diff
@@ -110,26 +111,40 @@ private void loadClassToCache(String key, Class clazz, String methodName) {
                 break;
             }
         }
+
         if (jsonMethod == null) {
             throw new SofaRpcRuntimeException(LogCodes.getLog(LogCodes.ERROR_METHOD_NOT_FOUND, clazz.getName(),
                 methodName));
         }
 
-        // parse request types
-        Type[] parameterTypes = jsonMethod.getGenericParameterTypes();
+        Type[] parameterTypes;
+        Type resType;
+        if (jsonMethod.getDeclaringClass() != clazz && !clazz.isInterface()) {
+            parameterTypes = new Type[1];
+            Type type = clazz.getGenericSuperclass();
+            ParameterizedType parameterizedType = (ParameterizedType) type;
```

Comment:
U may not assume that any type here is ParameterizedType. 

Say if `Class c extends B implement A`, B may not always be a ParameterizedType

Reply:
The new pull request fixes the parent object non-ParameterizedType conversion problem